.github/workflows/minimal-tests.yml

```diff
@@ -14,30 +14,37 @@ permissions:
 jobs:
   Check:
 
-    runs-on: ubuntu-latest
-    env:
-      AWS_DEFAULT_REGION: us-east-1
     strategy:
+      fail-fast: false
       matrix:
         python-version: [3.8]
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+
+    runs-on: ${{ matrix.platform }}
 
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Upgrade Pip
+        run: python -m pip install --upgrade pip
+      - name: Install Poetry
+        run: python -m pip install poetry
       - name: Install Requirements
         run: |
-          python -m pip install --upgrade pip
-          python -m pip install poetry
-          poetry config virtualenvs.create false --local
+          poetry config virtualenvs.in-project true
+          poetry config virtualenvs.path .venv
           poetry install -vvv
       - name: Test Metadata
-        run: pytest tests/unit/test_metadata.py
+        run: poetry run pytest tests/unit/test_metadata.py
       - name: Test Session
-        run: pytest tests/unit/test_session.py
+        run: poetry run pytest tests/unit/test_session.py
       - name: Test Utils
-        run: pytest tests/unit/test_utils.py
+        run: poetry run pytest tests/unit/test_utils.py
       - name: Test Moto
-        run: pytest -n 4 tests/unit/test_moto.py
+        run: poetry run pytest -n 4 tests/unit/test_moto.py
```

Comment:
I had to make all of these run in a virtual environment because Windows didn't like us upgrading the root `msgpack`.